modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureClientProvider.java

```diff
@@ -259,38 +264,136 @@ protected void doStop() {
     @Override
     protected void doClose() {}
 
-    private static final class SuccessfulRequestTracker implements HttpPipelinePolicy {
-        private static final Logger logger = LogManager.getLogger(SuccessfulRequestTracker.class);
+    static class RequestMetrics {
+        private volatile long timeToResponseInMillis = -1;
+        private volatile int requestCount;
+        private volatile int errorCount;
+        private volatile int throttleCount;
+        private volatile int statusCode;
```

Comment:
These are volatile because they're accessed by multiple threads, but we increment them non-atomically because they should only be modified by one thread at a time.